### PR TITLE
stdout output

### LINKED ISSUE #218: Accept stdout as file output
Requested here: https://discuss.elastic.co/t/is-it-possible-to-write-to-dev-stdout-using-file-output/32935

We need to add a special behavior if the file is /dev/stdout.

### PR DESCRIPTION
**Pull request by urso**
Add console output plugin writing to stdout.

resolves #218

### CODE REVIEW

File: etc/libbeat.yml [hunk 1]
```diff
@@ -153,6 +153,14 @@ output:
     # is 7 files.
     #number_of_files: 7
 
+  ### Console output
+  # console:
+    # Enabling console output
+    #enabled: false
+
+    # indent json
+    #indent: false
```
Comment (ruflin) [inline etc/libbeat.yml]: With indent you mean the same as ?pretty in elasticsearch?

Comment (urso) [inline etc/libbeat.yml]: oh, yes. Used nomenclator from golang doc

File: outputs/console/console.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		if err != nil {
+			return err
+		}
+
+		written += n
+	}
+	return nil
+}
+
+func (c *console) PublishEvent(
```
Comment (ruflin) [inline outputs/console/console.go]: Isn't also PublishEvents required for all outputs?

Comment (urso) [inline outputs/console/console.go]: no, the outputs plugin factory will wrap a plugin if it only implements Outputer but not BulkOutputer. See file output plugin

File: outputs/console/console.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		jsonEvent, err = json.Marshal(event)
+	}
+	if err != nil {
+		logp.Err("Fail to convert the event to JSON: %s", err)
+		outputs.SignalCompleted(s)
+		return err
+	}
+
+	if err = writeBuffer(jsonEvent); err != nil {
+		goto fail
```
Comment (ruflin) [inline outputs/console/console.go]: :-1:

**Comment by ruflin** [comment]
LGMT. Can you squash?

**Comment by urso** [comment]
done

**Comment by mrkschan** [comment]
:heart: :beers: :+1:

**Comment by mrkschan** [comment]
Any reason not to load `console` output? https://github.com/elastic/libbeat/blob/498121eceef2197840315850eb4c93c76d75ef4d/publisher/publish.go#L19 @urso

**Comment by urso** [comment]
oh, damit

**Pull request merged**